### PR TITLE
Add an aspect for DogService + modify Loggable annotation

### PR DESCRIPTION
**Pull request by gorshkovayulia**

### CODE REVIEW

**Comment by gorshkovayulia** [comment]
@ctapobep do you know what's wrong with my dogServiceAspect?
`WARNING: Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dogServiceAspect' defined in class path resource [controllers-context.xml]: BeanPostProcessor before instantiation of bean failed; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.aop.aspectj.AspectJPointcutAdvisor#0': Cannot create inner bean '(inner bean)#132e0cc' of type [org.springframework.aop.aspectj.AspectJAroundAdvice] while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name '(inner bean)#132e0cc': Cannot create inner bean '(inner bean)#1fc32e4f' of type [org.springframework.aop.config.MethodLocatingFactoryBean] while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name '(inner bean)#1fc32e4f': Initialization of bean failed; nested exception is java.lang.AbstractMethodError`

**Comment by ctapobep** [comment]
> Initialization of bean failed; nested exception is java.lang.AbstractMethodError

Seems like you're [initializing an abstract class](https://docs.oracle.com/javase/7/docs/api/java/lang/AbstractMethodError.html)?

File: src/main/resources/controllers-context.xml [hunk 1]
```diff
@@ -3,20 +3,35 @@
...
+       https://www.springframework.org/schema/aop/spring-aop.xsd"
+       default-autowire="constructor">
+
+    <bean id="dogServiceAspect" class="com.example.dog.utils.DogServiceAspect" />
+
+    <aop:config>
+        <aop:aspect ref="dogServiceAspect">
+            <aop:around
+                    pointcut="@annotation(com.example.dog.utils.CustomTransactional)"
+                    method="addTransactionalSupport" arg-names="pjp,jp"/>
```
Comment (ctapobep) [inline src/main/resources/controllers-context.xml:23]: Are these arg names for the pointcut or for the advice?

Comment (gorshkovayulia) [reply src/main/resources/controllers-context.xml:23]: For advice

**Comment by gorshkovayulia** [comment]
> > Initialization of bean failed; nested exception is java.lang.AbstractMethodError
> 
> Seems like you're [initializing an abstract class](https://docs.oracle.com/javase/7/docs/api/java/lang/AbstractMethodError.html)?

AbstractMethodError may be caused by the mismatch in the library version

**Pull request merged**